### ISSUE TITLE
Result is different from the paper for the mosi dataset

### ISSUE THREAD

**Issue by anuragrawal2024** [declare-lab/Multimodal-Infomax#16]
Hi @Clement25 ,

I trained the model on the mosi dataset and got the following numbers:
![image](https://github.com/declare-lab/Multimodal-Infomax/assets/140737044/d50e8491-a50b-4a4f-a6df-2c5db0c2e350)

They look different than ones reported in the paper:
![image](https://github.com/declare-lab/Multimodal-Infomax/assets/140737044/034713da-91b6-4882-9803-14ca002ef60e)

Would you know what could have caused these differences? Thanks for your help!

**Comment by sprog1** [comment]
I've encountered a similar issue, although my training was conducted on the Mosei database. In my case, the test set loss reached its minimum around the second epoch, but the training set loss stopped decreasing around the tenth epoch. However, ACC-2 and ACC-7 both seem to plateau at around 55%. I was wondering if you managed to resolve this issue and would greatly appreciate your insights.

**Comment by sprog1** [comment]
I've encountered a similar issue, although my training was conducted on the Mosei database. In my case, the test set loss reached its minimum around the second epoch, but the training set loss stopped decreasing around the tenth epoch. However, ACC-2 and ACC-7 both seem to be two points smaller than the results in the paper. And MAE is also two points higher than the results in the paper.I was wondering if you managed to resolve this issue and would greatly appreciate your insights.Thank you very much for your assistance.

**Comment by anuragrawal2024** [comment]
@sprog1, I have not been able to resolve it and am still waiting to hear from the authors.

**Comment by Clement25** [comment]
Hi, @utility-aagrawal and @sprog1 . Could you please try the following value settings and see how it works? Since from my experiment record they show similar good performance as the default settings.

lr_main=2e-4, lr_mmllb=5e-3, alpha=0.1, beta=0.3
lr_main=1e-3, lr_mmllb=5e-3, alpha=0.3, beta=0.1,
lr_main=1e-3, lr_mmllb=5e-3, alpha=0.3, beta=0.1 (default)
lr_main=2e-4, lr_mmllb=1e-3, alpha=0.3, beta=0.3

BTW, which type of GPU do you use? My implementation was on RTXA8000, using different types of GPUs may produce some slight variance.

**Comment by yangmiemiemie1** [comment]
对于MOSI数据集
使用作者提供代码python main.py --dataset mosi --contrast运行
运行结果为
![image](https://github.com/declare-lab/Multimodal-Infomax/assets/115394702/1ab657f1-d3e1-4b58-a495-2992a03c9d6b)


去掉对比学习 运行代码python main.py --dataset mosi 
![MOSI](https://github.com/declare-lab/Multimodal-Infomax/assets/115394702/68983a9e-54c8-45db-a109-c28dea3ce957)


为什么后面这面运行效果要更好一些

**Comment by Clement25** [comment]
> 对于MOSI数据集 使用作者提供代码python main.py --dataset mosi --contrast运行 运行结果为 ![image](https://user-images.githubusercontent.com/115394702/275498409-1ab657f1-d3e1-4b58-a495-2992a03c9d6b.png)
> 
> 去掉对比学习 运行代码python main.py --dataset mosi ![MOSI](https://user-images.githubusercontent.com/115394702/275498808-68983a9e-54c8-45db-a109-c28dea3ce957.png)
> 
> 为什么后面这面运行效果要更好一些

Hi, you need to parse the settings of hyper-paramters as the paper provides. The default setting is different from optimal setting for MOSI. For example, the default lr_main is 1e-3 but best setting for MOSI is 5e-3.

**Comment by yangmiemiemie1** [comment]
您好，超参数已经按照您提供的表格中进行了相应的设置，但是实验效果并不理想

---- Replied Message ----
| From | Wei Han ***@***.***> |
| Date | 10/20/2023 10:12 |
| To | declare-lab/Multimodal-Infomax ***@***.***> |
| Cc | yangmiemiemie1 ***@***.***>,
Comment ***@***.***> |
| Subject | Re: [declare-lab/Multimodal-Infomax] Result is different from the paper for the mosi dataset (Issue #16) |

对于MOSI数据集 使用作者提供代码python main.py --dataset mosi --contrast运行 运行结果为

去掉对比学习 运行代码python main.py --dataset mosi

为什么后面这面运行效果要更好一些

Hi, you need to parse the settings of hyper-paramters as the paper provides. The default setting is different from optimal setting for MOSI. For example, the default lr_main is 1e-3 but best setting for MOSI is 5e-3.

**Comment by Clement25** [comment]
Could you please try the hps I provided in
https://github.com/declare-lab/Multimodal-Infomax/issues/16#issuecomment-1722775295
